File: bumperbot_ws/src/bumperbot_localization/nodes/imu_republisher_node.py
#!/usr/bin/env python3
import rospy
from sensor_msgs.msg import Imu


frame_id = ""

def imuCallback(imu):
    imu.header.frame_id = frame_id
    imu_pub.publish(imu)

if __name__== '__main__':
    rospy.init_node('imu_republisher_node')
    frame_id = rospy.get_param("~frame_id")

    imu_pub = rospy.Publisher("imu_ekf", Imu, queue_size=10)
    imu_sub = rospy.Subscriber("imu", Imu, imuCallback)

    rospy.spin()

File: bumperbot_ws/src/bumperbot_localization/nodes/imu_republisher_node.cpp
#include <ros/ros.h>
#include <sensor_msgs/Imu.h>


std::string frame_id = "";
ros::Publisher imu_pub;

void imuCallback(const sensor_msgs::Imu &imu)
{
    sensor_msgs::Imu new_imu;
    new_imu = imu;
    new_imu.header.frame_id = frame_id;
    imu_pub.publish(new_imu);
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "imu_republisher_node");
    ros::NodeHandle nh;
    ros::NodeHandle pnh("~");
    pnh.getParam("frame_id", frame_id);

    imu_pub = nh.advertise<sensor_msgs::Imu>("imu_ekf", 10);
    ros::Subscriber imu_sub = nh.subscribe("imu", 1000, imuCallback);


    ros::spin();
    return 0;
}